test/move_only_string.hpp

```diff
@@ -0,0 +1,78 @@
+// cmcstl2 - A concept-enabled C++ standard library
+//
+//  Copyright Casey Carter
+//  Copyright Eric Niebler
+//
+//  Use, modification and distribution is subject to the
+//  Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at
+//  http://www.boost.org/LICENSE_1_0.txt)
+//
+// Project home: https://github.com/caseycarter/cmcstl2
+//
+#ifndef STL2_MOVE_ONLY_STRING_HPP
+#define STL2_MOVE_ONLY_STRING_HPP
+
+#include <cstring>
+#include <stl2/detail/swap.hpp>
+
+namespace cmcstl2_test {
+	namespace ranges = __stl2;
+	// a simple type to test move semantics
+	struct move_only_string {
+		move_only_string(char const* sz = "") noexcept
+		: sz_(sz)
+		, length_(std::strlen(sz))
+		{}
+		move_only_string(move_only_string&& that) noexcept
+		: sz_(ranges::exchange(that.sz_, ""))
+		, length_(ranges::exchange(that.length_, 0))
+		{}
+		move_only_string(move_only_string const&) = delete;
+		move_only_string& operator=(move_only_string&& that) noexcept
+		{
+			*this = ranges::exchange(that, move_only_string{});
```

Comment:
No change requested, just pointing out for future reference that `exchange`'s second template parameter defaults to the first, so you could equivalently write `*this = ranges::exchange(that, {});` here. `exchange(foo, {})` is very nicely idiomatic for "put `foo` in a value-initialized state and return its prior state".